yt_dlp/extractor/vrt.py

```diff
@@ -1,82 +1,215 @@
-        if description == '…':
-            description = None
-        timestamp = unified_timestamp(self._html_search_meta(
-            'article:published_time', webpage))
+        display_id = self._match_id(url)
+
+        model_json = self._download_json(f'{url.strip("/")}.model.json', display_id,
+                                         'Downloading asset JSON', 'Unable to download asset JSON')
+        details = model_json.get('details')
+        actions = details.get('actions')
+        title = details.get('title')
+        episode_publication_id = actions[2].get('episodePublicationId')
+        episode_video_id = actions[2].get('episodeVideoId')
+        video_id = f'{episode_publication_id}${episode_video_id}'
+        description = details.get('description')
+        episode = details.get('data').get('episode')
+        display_id = episode.get('name')
+        timestamp = parse_iso8601(episode.get('onTime').get('raw'))
+        upload_date = strftime('%Y%m%d', gmtime(timestamp))
+        series_info = details.get('data')
+        series = series_info.get('program').get('title')
+        season = series_info.get('season').get('title').get('value')
+        season_number = series_info.get('season').get('title').get('raw')
+        season_id = series_info.get('season').get('id')
+        episode = series_info.get('episode').get('number').get('value')
+        episode_number = series_info.get('episode').get('number').get('raw')
+        episode_id = series_info.get('episode').get('id')
+
+        video_info = None
+        vrtnutoken = ""
+
+        if self._authenticated:
+            vrtnutoken = self._download_json('https://token.vrt.be/refreshtoken',
+                                             video_id, note='refreshtoken: Retrieve vrtnutoken',
+                                             errnote='refreshtoken failed')['vrtnutoken']
+
+        headers = self.geo_verification_headers()
+        headers.update({'Content-Type': 'application/json; charset=utf-8'})
+        vrtPlayerToken = self._download_json(
+            '%s/tokens' % self._REST_API_BASE_TOKEN, video_id,
+            'Downloading token', headers=headers, data=json.dumps({
+                'identityToken': vrtnutoken
+            }).encode('utf-8'))['vrtPlayerToken']
+
+        video_info = self._download_json(
+            '%s/media-items/%s' % (self._REST_API_BASE_VIDEO, video_id),
+            video_id, 'Downloading video JSON', query={
+                'vrtPlayerToken': vrtPlayerToken,
+                'client': 'vrtnu-web@PROD',
+            }, expected_status=400)
+        if 'title' not in video_info:
+            code = video_info.get('code')
+            if code == 'AUTHENTICATION_REQUIRED':
+                self.raise_login_required()
+            elif code == 'INVALID_LOCATION':
+                self.raise_geo_restricted(countries=['BE'])
+            raise ExtractorError(video_info.get('message') or code, expected=True)
```

Comment:
I found some more error codes while testing, added them:
```suggestion

        if 'title' not in video_info:
            code = video_info.get('code')
            if code in ('AUTHENTICATION_REQUIRED', 'CONTENT_IS_AGE_RESTRICTED'):
                self.raise_login_required(code)
            elif code in ('INVALID_LOCATION', 'CONTENT_AVAILABLE_ONLY_IN_BE'):
                self.raise_geo_restricted(countries=['BE'])
            elif code == 'CONTENT_AVAILABLE_ONLY_FOR_BE_RESIDENTS_AND_EXPATS':
                if not self._authenticated:
                    self.raise_login_required(code)
                self.raise_geo_restricted(countries=['BE'])
            raise ExtractorError(code, expected=True)
```